src/js/app.js

```diff
@@ -0,0 +1,137 @@
+import capitalize from 'capitalize';
+
+new Vue({
+  el: '#app',
+
+  data: {
+    token: null,
+    category: null,
+    filter: null,
+    categories: [],
+    entries: [],
+    requests: {
+      left: 0,
+      total: 0,
+      reset: 0
+    },
+    loading: false,
+    interval: -1,
+    resetTimestamp: '',
+    error: null
+  },
+  computed: {
+    filteredEntries() {
+      const self = this;
+      if (this.entries.length === 0) return [];
+      return this.entries.filter(entry => entry.name.indexOf(self.filter) !== -1);
+    },
+    selectedCategory() {
+      return this.categories.find(el => el.id === this.category);
+    },
+    shouldShowList() {
+      return this.error === null && this.entries.length > 0;
+    },
+    loadingText() {
+      if (this.error) return this.error.toString();
+      return 'Loading...';
+    }
+  },
+
+  methods: {
+    transformName(name) {
+      return capitalize.words(name)
+        .replace('.Json', '')
+        .split('-').join(' / ');
+    },
+    transformResponse(data) {
+      return data.map(el => {
+        return {
+          id: el.name,
+          name: this.transformName(el.name),
+          url: el.download_url
+        };
+      });
+    },
+    loadCategories() {
+      const url = 'https://api.github.com/repos/axelrindle/collections/contents/data?ref=convert-data';
+      const self = this;
+      this.loading = true;
+      this.error = null;
+      axios.get(url)
+        .then(response => {
+          self.requests = {
+            left: response.headers['x-ratelimit-remaining'],
+            total: response.headers['x-ratelimit-limit'],
+            reset: response.headers['x-ratelimit-reset']
+          };
+          self.categories = self.transformResponse(response.data);
+          return self.loadEntries();
+        })
+        .catch(err => {
+          console.error(err);
+          self.error = err;
+        });
+    },
+    loadEntries() {
+      const self = this;
+      this.loading = true;
+      this.error = null;
+      axios.get(this.selectedCategory.url)
```

Comment:
Instead of reading from `http://raw.githubusercontent.com/`, use the deployed version from github pages after the merge.